src/Lookup/LookupDialog.tsx

```diff
@@ -1,26 +1,60 @@
-import PropTypes from "prop-types";
 import * as React from "react";
 import { createPortal } from "react-dom";
 import Draggable from "react-draggable";
-import { isDescendant } from "../../utils";
 import _ from "lodash";
 import classNames from "classnames";
+import { isDescendant } from "../utils";
 
-export default class LookupDialog extends React.Component {
+export interface LookupDialogProps {
+	/**
+	 * Callback executed when a dialog is closed
+	 */
+	close: (event: any) => any;
+	/**
+	 * Abolute or fixed positioning for a dialog
+	 */
+	position: { x: number; y: number };
+	/**
+	 * Enabled drag n drop for a diealog
+	 */
+	isDraggable?: boolean;
+	/**
+	 * Labelled by Id
+	 */
+	ariaLabelledBy?: string;
+	/**
+	 * Draggable handle selector
+	*/
+	dragHandle?: string;
+	/**
+	 * Portal target id
+	 */
+	portalTarget?: string;
+	/**
+	 * Popup container class name
+	 */
+	className?: string;
+	/** 
+	 * clickoutside props 
+	 */
+	[key: string]: any;
+}
+
+export default class LookupDialog extends React.Component<LookupDialogProps> {
 	static defaultProps = {
 		dragHandle: ".popup-container",
 		className: "popup-container dialog"
 	};
-	draggableContainer = React.createRef();
-	containerRef = React.createRef();
+	draggableContainer = React.createRef() as React.MutableRefObject<HTMLElement>;
```

Comment:
Why can't you use `React.createRef<HTMLElement>()` here?

Reply:
React.RefObject<T> is defined as immutable type.

Reply:
Then why do you need to use ref at all if you don't pass it to the `ref` prop? It can simply be a variable of type `HTMLElement` - it's a small refactoring that will eliminate such typing workarounds

Reply:
Make it a var of type HTMLElement if it's not passed to `ref` prop

Reply:
Done